src/common/predicates.js

```diff
@@ -2,6 +2,10 @@ const { compose, isNil, isEmpty, or, not, equals, type } = require('ramda')
 
 const isNilOrEmpty = (string) => or(isNil(string), isEmpty(string))
 
+const isNilOrEmptyOrUndefined = (value) => or(isUndefined(value), isNilOrEmpty(value))
+
+const notIsUndefinedOrNullOrEmpty = compose(not, isNilOrEmptyOrUndefined)
```

Comment:
how about `hasNonEmptyValue`?

Reply:
That could be confusing when passing in a list. To me that would sound like we're checking if the list contains any empty elements.